frontend/src/lib/ForwardMessageCache.test.ts

```diff
@@ -119,17 +121,40 @@ test("caches messages correctly", async () => {
     msg3.metadata.deltaPath = [2]
   }
   const ref = createRefMsg(msg3)
-  const unreferenced = await cache.processMessagePayload(ref)
+  const encodedRefMsg = ForwardMsg.encode(ref).finish()
+  const unreferenced = await cache.processMessagePayload(ref, encodedRefMsg)
   expect(getCachedMessage(ref.hash)).toBeUndefined()
   expect(unreferenced).toEqual(msg3)
 
   // Test that our uncached messages are copies
   expect(unreferenced).not.toBe(msg3)
 })
 
+test("caches messages as a deep copy", async () => {
+  const { cache, getCachedMessage } = createCache()
+
+  const msg = createForwardMsg("Cacheable", true)
+  const encodedMsg = ForwardMsg.encode(msg).finish()
+
+  await cache.processMessagePayload(msg, encodedMsg)
+
+  // Check if message is correctly cached
+  expect(getCachedMessage("Cacheable")).toEqual(msg)
+  expect(getCachedMessage("Cacheable")?.metadata?.deltaPath).toEqual([])
+
+  // Modify the message content
+  if (msg.metadata) {
+    msg.metadata.deltaPath = [10]
+  }
+
+  // Check that it does not impact the cached message
+  expect(getCachedMessage("Cacheable")?.metadata?.deltaPath).not.toEqual([10])
```

Comment:
A bit of a nit, but I think this would be closer to the intention of the test if this `expect` tests that the original message has its original value, as opposed to _not_ having the modified value. How about initializing the original msg with a non-default value (`msg.metadata.deltaPath = [5]` or whatever), and testing that it has that value here?

Reply:
I think ideally we'd modify the metadata AND some non-metadata for this test, since those two bits of data are handled differently, and we want to make sure neither the metadata nor the actual message payload are modified.

Reply:
Good point! Added `delta` data as well and changed it to check the original values 👍